### PR TITLE
Add additional checks and tests

### PR DESCRIPTION
**Pull request by ka-sarthak**
- [x] add checks in the function to handle None for value and units
- [x] add unit tests for the function

### CODE REVIEW

**Review by hampusnasstrom — COMMENTED** [review]
Try if my simplification works, maybe also add to the descriptions that it accepts pint.Quantity

File: src/nomad_measurements/utils.py [hunk 1]
```diff
@@ -126,6 +126,12 @@ def to_pint_quantity(value: Any=None, unit: str=None) -> Any:
...
     '''
-    if isinstance(value, str):
+    if isinstance(value, str) or value is None:
         return value
+    if unit is None:
+        if isinstance(value, ureg.Quantity):
+            return value
+        return value * ureg.dimensionless
+    if isinstance(value, ureg.Quantity):
+        return value.to(unit)
```
[the remark this answers is not in the document]
Comment (ka-sarthak) [reply src/nomad_measurements/utils.py:136]: Indeed, looks good!

**Comment by ka-sarthak** [comment]
Thanks for the suggestions; incorporated them. Please have a look

**Review by hampusnasstrom — APPROVED** [review]
Great, thanks!

**Pull request merged**